test/index.spec.js

```diff
@@ -207,6 +221,21 @@ describe('variants', () => {
     expect(addr).to.have.property('buffer')
     expect(addr.toString()).to.equal(str)
   })
+
+  it('ip6 + tcp + websockets + ipfs', () => {
+    const str = '/ip6/2001:8a0:7ac5:4201:3ac9:86ff:fe31:7095/tcp/8000/websockets/ipfs/QmcgpsyWgH8Y8ajJz1Cu72KnS5uo2Aa2LpzU7kinSupNKC'
```

Comment:
Can we have a test where we 'decapsulate' the ipfs part of a multiaddr like this and another test where we 'decapsulate' the websockets part, while checking the rest remains?